packages/account/src/Sections/Profile/FinancialAssessment/financial-assessment.jsx

```diff
@@ -194,20 +194,11 @@ class FinancialAssessment extends React.Component {
                             (this.props.is_financial_account || this.props.is_trading_experience_incomplete) &&
                             !this.props.is_svg;
 
-                        const needs_financial_assessment =
-                            this.props.is_financial_information_incomplete ||
-                            this.props.is_high_risk ||
-                            has_trading_experience ||
-                            this.props.is_financial_account;
                         this.setState({ has_trading_experience });
 
                         if (data.error) {
                             this.setState({ api_initial_load_error: data.error.message });
                             return;
-                        } else if (!needs_financial_assessment) {
-                            // Additional layer of error handling if non high risk user somehow manages to reach FA page
-                            // need to redirect to default route for account to prevent app crash
-                            this.props.history.push(routes.personal_details);
```

Comment:
We still need this condition to prevent app from crashing in case user manages to reach FA page!

Reply:
Thank you a lot for the review.
Do I understand the purpose of the line 68 in account.jsx correctly? It contains a condition making the app redirect a user back to trader if he/she opens FA page and doesn't need_financial_assessment (e.g. filled up already, etc.). My task is to enable user to view and edit his FA after it's filled up which means that a user should not be redirected anywhere from this page as I understand.
I can also see that needs_financial_assessment from client-store.js is used in 4 other files besides account.jsx. That's why I believe that adding additional conditions to needs_financial_assessment in client-store.js might affect those 4.
My decision to remove needs_financial_assessment from account.jsx is based on understanding that it's used in account.jsx for only 2 purposes that I want to eliminate according to my task:
1. to redirect a user back to trader if he/she opens FA page and doesn't need_financial_assessment, e.g. filled up already, etc.,
2. to disable the Financial Assessment in the list of Account Settings if a user doesn't need_financial_assessment, e.g. filled up already, etc.
Also, if you try to reach FA page from a high-risk account by changing the URL now, the app won't crash anymore because I removed the line 68 in account.jsx. So there is hardly any reason to redirect a user to Personal Details from FA now.
Or Is there any reason for the app to crash that I don't see? While I tested it, it didn't crash.
Please tell me how I should proceed.

Reply:
Your logic is quite correct, but you need to consider that your task only considers CR accounts but from your approach in this PR, you are affecting other accounts as well.